### PR TITLE
feat: Bridge Command Events

### PR DESCRIPTION
**Pull request by Om1609**
## Summary

Adds `on_bridge_command`, `on_bridge_command_completion` and `on_bridge_command_error`

## Information

- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed).
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting,
      examples, ...).

## Checklist

- [x] I have searched the open pull requests for duplicates.
- [x] If code changes were made then they have been tested.
  - [x] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why.

### CODE REVIEW

**Review by JustaSqu1d — COMMENTED** [review]
Add a changelog entry.

File: CHANGELOG.md [hunk 1]
```diff
@@ -34,6 +34,9 @@ These changes are available on the `master` branch, but have not yet been releas
 - Added new raw events: `raw_member_remove`, `raw_thread_update`, and
   `raw_thread_member_remove`.
   ([#1880](https://github.com/Pycord-Development/pycord/pull/1880))
+- Added new events `on_bride_command`, `on_bridge_command_completion` and
```
Comment (JustaSqu1d) [inline CHANGELOG.md:37]: Use an oxford comma and fix spelling.
```suggestion
- Added new events `on_bridge_command`, `on_bridge_command_completion`, and
```

File: docs/ext/bridge/api.rst [hunk 2]
```diff
@@ -40,6 +40,43 @@ AutoShardedBot
 .. autoclass:: discord.ext.bridge.AutoShardedBot
     :members:
 
+Event Reference
+---------------
+
+These events function similar to :ref:`the regular events <discord-api-events>`, except they
+are custom to the bridge extension module.
+
+.. function:: discord.ext.bridge.on_command_error(ctx, error)
```
Comment (JustaSqu1d) [inline docs/ext/bridge/api.rst:49]:
```suggestion
.. function:: discord.ext.bridge.on_bridge_command_error(ctx, error)
```

File: docs/ext/bridge/api.rst [hunk 3]
```diff
@@ -40,6 +40,43 @@ AutoShardedBot
...
+    An error handler that is called when an error is raised
+    inside a command either through user input error, check
+    failure, or an error in your own code.
+
+    :param ctx: The invocation context.
+    :type ctx: :class:`.Context`
+    :param error: The error that was raised.
+    :type error: :class:`.CommandError` derived
+
+.. function:: discord.ext.bridge.on_command(ctx)
```
Comment (JustaSqu1d) [inline docs/ext/bridge/api.rst:60]:
```suggestion
.. function:: discord.ext.bridge.on_bridge_command(ctx)
```

File: docs/ext/bridge/api.rst [hunk 4]
```diff
@@ -40,6 +40,43 @@ AutoShardedBot
...
+
+    An event that is called when a command is found and is about to be invoked.
+
+    This event is called regardless of whether the command itself succeeds via
+    error or completes.
+
+    :param ctx: The invocation context.
+    :type ctx: :class:`.Context`
+
+.. function:: discord.ext.bridge.on_command_completion(ctx)
```
Comment (JustaSqu1d) [inline docs/ext/bridge/api.rst:70]:
```suggestion
.. function:: discord.ext.bridge.on_bridge_command_completion(ctx)
```

File: docs/ext/bridge/api.rst [hunk 5]
```diff
@@ -40,6 +40,43 @@ AutoShardedBot
...
+
+    An event that is called when a command has completed its invocation.
+
+    This event is called only if the command succeeded, i.e. all checks have
+    passed and the user input it correctly.
```
Comment (JustaSqu1d) [inline docs/ext/bridge/api.rst:75]: Subject-verb agreement
```suggestion
    passed and users input it correctly.
```

File: discord/ext/bridge/bot.py [hunk 6]
```diff
@@ -106,6 +117,54 @@ def decorator(func) -> BridgeCommandGroup:
 
         return decorator
 
+    async def invoke(self, ctx: ExtContext | BridgeExtContext):
+        if ctx.command is not None:
+            self.dispatch("command", ctx)
+            if br_cmd := isinstance(ctx.command, BridgeExtCommand):
+                self.dispatch("bridge_command", ctx)
```
Comment (JustaSqu1d) [inline discord/ext/bridge/bot.py:124]: Should the text-based commands be called as well for consistency with application commands?

**Review by JustaSqu1d — CHANGES_REQUESTED** [review]

**Review by JustaSqu1d — APPROVED** [review]

**Review by Lulalaby — APPROVED** [review]

**Review by Middledot — APPROVED** [review]

**Pull request merged**